src/status_im2/contexts/share/events.cljs

```diff
@@ -4,12 +4,18 @@
             [quo2.foundations.colors :as colors]))
 
 (rf/defn copy-text-and-show-toast
+  "This function will copy the value provided in arguments and
+  first close all open toast notifications and then show a toast
+  notification indicating a successful copy. This prevents multiple
+  toast notifications spamming on the screen"
   {:events [:share/copy-text-and-show-toast]}
-  [cofx text-to-copy post-copy-message]
+  [{:keys [db] :as cofx} {:keys [text-to-copy post-copy-message]}]
   (rf/merge cofx
             {:copy-to-clipboard text-to-copy}
+            (toasts/close :successful-copy-toast-message)
             (toasts/upsert
              {:icon           :correct
+              :id             :successful-copy-toast-message
```

Comment:
Since now we're passing the `id`, why do we need to call `toasts/close` @siddarthkay? I thought `upsert` with the same ID would just replace the existing one, so there would be no need to close it first.

Reply:
@ilmotta : fair enough, i just saw your explanation on the issue, yeah the `toasts/close` was an overkill on my part.